src/test/basics/tagged_integer_test.cpp

```diff
@@ -52,17 +52,17 @@ class tagged_integer_test
         "TagUInt3 should be constructible using a std::uint64_t");
 
     // Check assignment of tagged_integers
-    static_assert (!std::is_assignable<TagUInt1, std::uint32_t>::value,
-        "TagUInt1 should not be assignable with a std::uint32_t");
+    static_assert (std::is_assignable<TagUInt1, std::uint32_t>::value,
```

Comment:
These are pretty big changes, i.e. they change how tagged_integer behaved pretty significantly.  (i.e. all the changes in this diff are now **allowing** conversions that weren't allowed before.